client/src/app/site/pages/organization/pages/accounts/pages/account-detail/components/account-add-to-meetings/account-add-to-meetings.component.ts

```diff
@@ -0,0 +1,164 @@
+
+@Component({
+    selector: `os-account-add-to-meetings`,
+    templateUrl: `./account-add-to-meetings.component.html`,
+    styleUrls: [`./account-add-to-meetings.component.scss`],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class AccountAddToMeetingsComponent extends BaseUiComponent implements OnInit, AfterViewInit {
+    public user: ViewUser | null = null;
+    public canManage = false;
+    public isOwnPage = false;
+
+    public meetingsSubject = new BehaviorSubject<ViewMeeting[]>([]);
+
+    public get warningMessage(): string {
+        if (!this.selectedMeetings?.length) {
+            return _(`No meetings have been selected.`);
+        }
+        if (!this.groupName) {
+            return _(`No group name has been entered.`);
+        }
+        return null;
+    }
+
+    public get meetingsObservable(): Observable<ViewMeeting[]> {
+        return this.meetingsSubject.asObservable();
+    }
+
+    public assignMeetingsForm = this.formBuilder.group({
+        group_name: [``],
+        meeting_ids: [[]]
+    });
+
+    public get selectedMeetings(): number[] {
+        return this.assignMeetingsForm.get(`meeting_ids`).value as number[];
+    }
+
+    public get groupName(): string {
+        return this.assignMeetingsForm.get(`group_name`).value as string;
+    }
+
+    public get hasResult(): boolean {
+        let dataLength = 0;
+        Object.keys(this.resultsTableDataSubjects).forEach(key => {
+            dataLength = dataLength + this.resultsTableDataSubjects[key].value.length;
+        });
+        return !!dataLength;
+    }
+
+    public resultsTableDataSubjects = {
+        succeeded: new BehaviorSubject<string[]>([]),
+        standard_group: new BehaviorSubject<string[]>([]),
+        nothing: new BehaviorSubject<string[]>([])
+    };
+
+    public get waitingForResults(): boolean {
+        return this._waitingForResults;
+    }
+
+    private _waitingForResults = false;
+
+    private userId: Id | null = null;
+
+    public constructor(
+        private operator: OperatorService,
+        private userController: UserControllerService,
+        private meetingController: MeetingControllerService,
+        private router: Router,
+        private osRouter: OpenSlidesRouterService,
+        private formBuilder: UntypedFormBuilder
+    ) {
+        super();
+    }
+
+    public ngOnInit(): void {
+        this.subscriptions.push(
+            this.osRouter.currentParamMap.subscribe(params => {
+                if (params[`id`]) {
+                    this.userId = +params[`id`];
+                    this.loadUser();
+                }
+            }),
+            this.meetingController
+                .getViewModelListObservable()
+                .subscribe(meetings => this.meetingsSubject.next(meetings))
+        );
+    }
+
+    public ngAfterViewInit(): void {
+        setTimeout(() => this.updatePermissions()); // Initial check
+    }
```

Comment:
This looks hacky. Maybe this is not needed (see comment regarding the permission variables above), but if it is, can you transform then into getters and let the change detection handle the updates so these helper functions can be removed?